xbmc/threads/platform/pthreads/ThreadImpl.cpp

```diff
@@ -38,6 +38,36 @@
 #include <signal.h>
 #include "utils/log.h"
 
+namespace XbmcThreads
+{
+  // ==========================================================
+  static pthread_mutexattr_t recursiveAttr;
+
+  static bool setRecursiveAttr()
+  {
+    static bool alreadyCalled = false; // initialized to 0 in the data segment prior to startup init code running
+    if (!alreadyCalled)
+    {
+      pthread_mutexattr_init(&recursiveAttr);
+      pthread_mutexattr_settype(&recursiveAttr,PTHREAD_MUTEX_RECURSIVE);
+#if !defined(__arm__) && !defined(TARGET_ANDROID)
```

Comment:
I don't think this restriction is correct. No reason why arm in general should be excluded

Reply:
It was removed for a good reason in the past - it caused deadlocks on arm platforms.
I'll see if I can find a link to discussion.

Reply:
Some testing here: https://github.com/xbmc/xbmc/pull/9430#issuecomment-200624811
Seems to cause a core to run at 100% CPU.

Reply:
Again, this is the way the class was.
https://github.com/xbmc/xbmc/pull/13454/files#diff-d85cd2654abbc311f859cdb2fbb7ce4aL41
I'm assuming this was originally your code.

Reply:
@popcornmix this is already 2 years ago. I suggest to retry, maybe the issue is resolved. Even if there are issues on Pi, there other playforms like iOS orAML should not suffer from priority inversion. @peak3d @Memphiz fyi

Reply:
We included the ``PTHREAD_PRIO_INHERIT`` in last night's Milhouse build and the 100% cpu issue wasn't observed. Perhaps the underlying issue has been resolved in the meantime?
As such I'm okay with remove the ``__arm__`` part of the ifdef so ``PTHREAD_PRIO_INHERIT`` is enabled on arms again.

Reply:
> Perhaps the underlying issue has been resolved in the meantime?

Probably, maybe older kernels did busy wait

Thanks for testing.